### PR TITLE
fix: wait for flush tasks to finish before cancelling them

### PR DESCRIPTION
**Pull request by ajcasagrande**
## Bug Fixes
* Improved data integrity when performing concurrent write operations
* Enhanced error handling and diagnostic logging during shutdown procedures
* Fixed potential resource leaks during application termination

## Tests
* Added comprehensive tests for concurrent write operation scenarios
* Added tests verifying buffer flushing and resource cleanup edge cases

### CODE REVIEW

File: src/aiperf/common/mixins/buffered_jsonl_writer_mixin.py [hunk 1]
```diff
@@ -92,21 +95,34 @@ async def _flush_buffer(self, buffer_to_flush: list[str]) -> None:
...
 
     @on_stop
     async def _close_file(self) -> None:
         """Flush remaining buffer and close the file handle (called automatically on shutdown)."""
+        # Wait for any pending flush tasks to complete
+        if self.tasks:
+            await asyncio.wait_for(
+                self.wait_for_tasks(),
+                timeout=Environment.SERVICE.TASK_CANCEL_TIMEOUT_SHORT,
+            )
```
[the remark this answers is not in the document]
Comment (debermudez) [reply src/aiperf/common/mixins/buffered_jsonl_writer_mixin.py]: @ajcasagrande is this something you think is valid and needs to be handled?

**Review by debermudez — APPROVED** [review]
approved. Timeout suggestion from coderabbit is not gating

File: src/aiperf/common/mixins/buffered_jsonl_writer_mixin.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ajcasagrande) [reply src/aiperf/common/mixins/buffered_jsonl_writer_mixin.py]: Yeah probably worth it just to be safe
